src/metrics/representation.py

```diff
@@ -340,11 +315,16 @@ def generator():
                                [6, 1]])
                     ].
             """
-            for i in range(0, len(queries), self.search_batch_size):
-                query_idxs = np.arange(i, min(i + self.search_batch_size, len(queries)))
+            for i in range(0, len(queries_idxs), self.search_batch_size):
+                idxs = np.arange(i, min(i + self.search_batch_size, len(queries_idxs)))
+
+                closest_dist, closest_idx = index.search(vectors[idxs], k=k)
+                # remove first element which is actually is it's classification dataset
```

Comment:
Remove the first retrieved element in case of classification dataset since it's a query itself